tests/iter.rs

```diff
@@ -35,9 +35,9 @@ fn iter_mut() {
     for particle in particles.iter_mut() {
         *particle.mass += 1.0;
     }
-    assert_eq!(particles.mass[0], 1.0);
-    assert_eq!(particles.mass[1], 1.0);
-    assert_eq!(particles.mass[2], 1.0);
+    assert_eq!(*particles.index(0).mass, 1.0);
```

Comment:
this changes because this API will no longer be supported with the new single-length code, right?

Reply:
Precisely. Stop relying on the underlying structure with `Vec`s.

Reply:
I see =). From a usability perspective, it could be nice to still enable direct indexing inside the vec members even after moving to a single length, although I don't know how we could do it. In the worst case the user have to create a slice instead

Reply:
I thought about using `Deref` and `DerefMut` traits to coerce into its slice counterparts (`Vec` already does it) but we have the same problem as `Index` and `IndexMut`: the trait forces you to return a regular reference.
I'd be cool to be able to hack the code to allow us to use it -maybe with some intermediate type that automatically coerces- but so far I've got no luck.

Reply:
Maybe add a trait and add the access as methods. This would mean no collisions with existing functions.

Reply:
I found a nice solution:
 - Add a level of indirection
 - Keep the vec methods in the "outer" struct
 - Add getters -`mass(&self)` and `mass_mut(&mut self)`- to the "inner" struct
 - Define `Deref` and `DerefMut` so we can use the getters without explicitly using the indirection

[Here is an example that I was playing with](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=bfd85fc1d8014f49a878e6f96fa54dc4)

Reply:
By the way, In this PR I'm not removing the access to the underlying vecs, I'm just making sure that the tests don't rely on it. A user using `stable` will still be able to use `particles.name[0]`, but in the single len version, they won't (at least for now).

Reply:
> I found a nice solution:

That could work nicely for the single length API =)